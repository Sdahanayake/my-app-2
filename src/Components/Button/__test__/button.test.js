import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../button';

import {render,cleanup} from '@testing-library/react';
import renderer from "react-test-renderer";
//import "jest-dom/extend-expect";

afterEach(cleanup);

it("renders without crashing ",()=>{
    const div=document.createElement("div");
    ReactDOM.render(<Button></Button>,div);



})

it("renders button correctly ",() =>{
   const {getByTestId}= render(<Button label="Click me"></Button>)
   expect(getByTestId('button')).toHaveTextContent("Click me");


})

it("renders button correctly",() =>{
    const{getByTestId} = render(<Button label="save" />);
    expect(getByTestId("button")).toHaveTextContent("save");
});

it("Matches snapshots 1",() =>{
    const tree =renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})
it("Matches snapshots 2",() =>{
    const tree =renderer.create(<Button label="Click me"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})