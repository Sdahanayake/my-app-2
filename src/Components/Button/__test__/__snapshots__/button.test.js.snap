// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Sumudu
  save
</div>
`;

exports[`Matches snapshots 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Sumudu
  Click me
</div>
`;
